presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ReplicateSemiJoinInDelete.java

```diff
@@ -23,41 +23,35 @@
 import com.facebook.presto.sql.planner.plan.SemiJoinNode;
 import com.facebook.presto.sql.planner.plan.SimplePlanRewriter;
 
-import java.util.Optional;
-
-import static com.facebook.presto.SystemSessionProperties.getJoinDistributionType;
+import static com.facebook.presto.sql.planner.plan.SemiJoinNode.DistributionType.REPLICATED;
 import static java.util.Objects.requireNonNull;
 
-public class DetermineSemiJoinDistributionType
+public class ReplicateSemiJoinInDelete
         implements PlanOptimizer
 {
     @Override
     public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator, WarningCollector warningCollector)
     {
         requireNonNull(plan, "plan is null");
         requireNonNull(session, "session is null");
+        requireNonNull(symbolAllocator, "symbolAllocator is null");
```

Comment:
nit: Only `plan` parameter is used. I don't think you need to check anything but `plan` with the `requireNonNull`.